### PR TITLE
feat(RPS-676): Added automated tests for story rendering

### PR DESCRIPTION
**Pull request by evrys**
Adds basic smoke tests for all stories in the storybook. Adapted from #129

Notable changes:
- `npm run test` tests all stories on local dev (requires an already running dev server)
- `npm run ci:test` builds the static production storybook and then runs all tests against that
- Github Actions now runs ci:test and checks all stories as part of the pipeline
- Special features for robot models
  - A robot model story is now created automatically for each `.glb` file in `public/models` using a custom storybook indexer
  - `GenericRobot` and the tests now validate that the glb file conforms to the expected structure (i.e. it has six `_Jxx` joint groups and a `_FLG` group)
  - Added a `postModelRender` callback on `SupportedRobot` which is called after the GLTF is processed and the r3f objects mounted
  - Model story test checks that `postModelRender` was called successfully for each robot

I decided not to do visual snapshot testing for now because I found it not quite reliable enough, but I think the smoke tests add quite a bit of value by themselves 🙂 

Additional story-specific checks can be added to the [play function](https://storybook.js.org/docs/writing-tests/component-testing) of a story as needed

Checklist:

- [x] Test fails if an error is thrown during rendering of React component
- [x] Test fails if a robot model fails to load (e.g. file missing)
- [x] Test fails if the joints of a robot could not be detected from the glb file
- [x] Test fails if the flange of a robot could not be detected from the glb file
- [x] Fix the UR5e

UR5e is missing a dedicated story currently (did it go missing?) but it's used in the example Robot story, so it throws the _FLG error (yay tests catching things)

- [x] Github Actions pipeline runs tests successfully

### CODE REVIEW

File: .github/workflows/check.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ jobs:
       - uses: actions/setup-node@v4
         with:
           node-version: "lts/*"
-          cache: 'npm'
+          cache: "npm"
```
Comment (bompo) [inline .github/workflows/check.yml:16]: did the auto formatter change?

File: src/components/robots/robotModelLogic.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    if (isJoint(node)) {
+      joints.push(node)
+    }
+
+    node.children.map(parseNode)
+  }
+
+  parseNode(gltf.scene)
+
+  if (!isSixJoints(joints)) {
```
Comment (bompo) [inline src/components/robots/robotModelLogic.ts:77]: Fine for now, just a note:
We should rename robots to motion group to also add external axis as well.

File: stories/robots/SupportedRobotScene.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import { OrbitControlsAround } from "./OrbitControlsAround"
+
+/**
+ * Renders a single robot model with orbit controls targeting it
+ * for storybook demonstration purposes
+ */
+export function SupportedRobotScene(
+  props: React.ComponentProps<typeof SupportedRobot>,
+) {
+  const flangeRef: React.RefCallback<Group> = useCallback((node) => {
```
Comment (bompo) [inline stories/robots/SupportedRobotScene.tsx:16]: This is also a nice example on how you would add a visual tool to the robot. We could add this to the general robot docs

**Review by bompo — APPROVED** [review]
Right call in not using visual snapshot tests. They would be too flaky.

File: .github/workflows/check.yml (hunk 1 above)
Comment (evrys) [reply .github/workflows/check.yml:16]: This file was added back in August, I think I didn't have the project vscode config set to format on save back then

File: src/components/robots/robotModelLogic.ts (hunk 2 above)
Comment (evrys) [reply src/components/robots/robotModelLogic.ts:77]: Good to remember! I was thinking about robots with more than 6 joints, but I figure it's pretty easy to change the validation once we do add those

File: stories/robots/SupportedRobotScene.tsx (hunk 3 above)
Comment (evrys) [reply stories/robots/SupportedRobotScene.tsx:16]: yuh! though this particular approach isn't very r3f friendly, it'd probably be better to find a way that fits into the component tree

**Review by vgerber — APPROVED** [review]
lgtm

**Comment by vgerber** [comment]
> Right call in not using visual snapshot tests. They would be too flaky.

I tested them on another branch and would agree. The models loaded quite unreliably

**Pull request merged**